src/providers/pkcs11/key_management.rs

```diff
@@ -133,6 +128,17 @@ impl Provider {
                 ));
                 Ok(Mechanism::RsaPkcsKeyPairGen)
             }
+            Type::EccKeyPair { curve_family } => {
+                pub_template.push(Attribute::Private(false.into()));
```

Comment:
Maybe the `Private` attribute can be set to false in both cases before the `match` in the public template.